flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/SplitAggregateRule.scala

```diff
@@ -303,7 +304,19 @@ class SplitAggregateRule extends RelOptRule(
             aggGroupCount + index + avgAggCount + 1,
             finalAggregate.getRowType)
           avgAggCount += 1
-          relBuilder.call(FlinkSqlOperatorTable.DIVIDE, sumInputRef, countInputRef)
+          // TODO
+          val equals = relBuilder.call(
+            FlinkSqlOperatorTable.EQUALS,
+            countInputRef,
+            relBuilder.getRexBuilder.makeBigintLiteral(JBigDecimal.valueOf(0)))
+          val falseT = relBuilder.call(FlinkSqlOperatorTable.DIVIDE, sumInputRef, countInputRef)
+          val trueT = relBuilder.cast(
+            relBuilder.getRexBuilder.constantNull(), aggCall.`type`.getSqlTypeName)
+          relBuilder.call(
+            FlinkSqlOperatorTable.IF,
```

Comment:
Flink planner supports `cast(null as bigint) / cast( 0 as bigint)` and get `null` result. If we supports this, can we remove this?

Reply:
We use SUM0 in this rule， which many run into zero / zero and throw / zero exception.

Reply:
This should be a regular case. Should it return null when avg don't have input count?

Reply:
Div 0 should return null instead of exception?

Reply:
I agree with @JingsongLi , we should never throw exception in blink planner. `null or zero / zero` should always return null, in that way, we can avoid to introduce this.

Reply:
In SQL 2013 Part 2 Section 6.27:
The dyadic arithmetic operators <plus sign>, <minus sign>, <asterisk>, and <solidus> (+, –, *, and /,
respectively) specify addition, subtraction, multiplication, and division, respectively. **If the value of a
divisor is zero, then an exception condition is raised: data exception — division by zero.**

And I think it's THE SplitAggretateRule owns the responsibility of return null(not throw exception).